### PR TITLE
bug[next]: Fix reverse operators in embedded

### PR DESCRIPTION
**Pull request by havogt**
The reverse operators where not swapping their arguments.

E.g. `1.0 - field` did `field - 1.0`.

### CODE REVIEW

File: src/gt4py/next/embedded/nd_array_field.py [hunk 1]
```diff
@@ -59,15 +65,20 @@ def _builtin_op(*fields: common.Field | core_defs.Scalar) -> NdArrayField:
                 if f.domain == domain_intersection:
                     transformed.append(xp.asarray(f.ndarray))
                 else:
-                    f_broadcasted = _broadcast(f, domain_intersection.dims)
+                    f_broadcasted = (
+                        _broadcast(f, domain_intersection.dims)
+                        if f.domain.dims != domain_intersection.dims
+                        else f
```
Comment (tehrengruber) [inline src/gt4py/next/embedded/nd_array_field.py:71]:
```suggestion
                        _broadcast(f, domain_intersection.dims)
                        if f.domain.dims != domain_intersection.dims
                        else f
```
I would expect this to be part of the broadcast operation. Can't we put it there?

File: tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py [hunk 2]
```diff
@@ -147,37 +167,48 @@ def test_where_builtin_with_tuple(nd_array_implementation):
...
     result = fbuiltins.where(cond_field, field_true, field_false)
     assert np.allclose(result[0].ndarray, expected0)
     assert np.allclose(result[1].ndarray, expected1)
 
 
-def test_binary_arithmetic_ops(binary_arithmetic_op, nd_array_implementation):
-    inp_a = [-1.0, 4.2, 42]
-    inp_b = [2.0, 3.0, -3.0]
-    inputs = [inp_a, inp_b]
+@pytest.mark.parametrize("lhs, rhs", [([-1.0, 4.2, 42], [2.0, 3.0, -3.0]), (1.0, [2.0, 3.0, -3.0])])
```
Comment (tehrengruber) [inline tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py:183]: Maybe leave a comment how the lhs-is-a-scalar case covers the `rsub`, ... cases, it took me a while to understand why this covers what you fix in this PR.

**Review by tehrengruber — APPROVED** [review]

File: src/gt4py/next/embedded/nd_array_field.py (hunk 1 above)
Comment (havogt) [reply src/gt4py/next/embedded/nd_array_field.py:71]: actually not sure why I cared for this optimization, done.

File: tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py (hunk 2 above)
Comment (havogt) [reply tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py:183]: done

**Pull request merged**